backend/application/advertisement.ts

```diff
@@ -0,0 +1,35 @@
+import { PrismaClient } from "@prisma/client";
+import crypto from 'crypto'
+const prisma = new PrismaClient()
+
+export class AdvertisementApplicationService {
+  async createAdvertisement(params: {
+    name: string,
+    url: string,
+    imageUrl: string,
+    author: string,
+    isShow: boolean,
+    startFrom: Date,
+    endAt: Date
```

Comment:
```suggestion
  async createAdvertisement(params: AdvertisementType
```

じゃないですかね？
せっかくAdvertisementTypeを外で定義しているのに使っていないのが勿体無いです。